kafka/consume.go

```diff
@@ -142,14 +178,26 @@ func (r *ConfluentConsumer) SetInitialOffset(offset int64) error {
 // FetchMessage implements Consumer interface.
 func (r *ConfluentConsumer) FetchMessage(ctx context.Context) (*kafkalib.Message, error) {
 	var err error
-	r.subscribeOnce.Do(func() {
-		r.log.WithField("kafka_topic", r.conf.Topic).Debug("Subscribing to Kafka topic")
-		r.rebalanceHandlerMutex.Lock()
-		defer r.rebalanceHandlerMutex.Unlock()
-		if err := r.c.Subscribe(r.conf.Topic, r.rebalanceHandler); err != nil {
-			err = errors.Wrap(err, "error subscribing to topic")
+	// if we are not reading from a partition, we subscribe
+	if r.partition < 0 {
+		r.subscribeOnce.Do(func() {
+			r.log.WithField("kafka_topic", r.conf.Topic).Debug("Subscribing to Kafka topic")
+			r.rebalanceHandlerMutex.Lock()
+			defer r.rebalanceHandlerMutex.Unlock()
+			if err := r.c.Subscribe(r.conf.Topic, r.rebalanceHandler); err != nil {
+				err = errors.Wrap(err, "error subscribing to topic")
+			}
+		})
+	} else {
+		parts := []kafkalib.TopicPartition{
+			kafkalib.TopicPartition{
+				Topic:     &r.conf.Topic,
+				Partition: r.partition,
+			},
 		}
-	})
+		err = r.c.Assign(parts)
```

Comment:
You wanna assign just once.

Reply:
I was thinking about this last night.  I think we should add the sync once and initialize a new sync.Once when we set a new partition with SetPartitionByKey method.  That would allow a single connection to read from multiple partitions

Reply:
i'd support having a `sync.Once` for this :+1: